Observers/SalesOrderInvoicePayObserver.php

```diff
@@ -5,53 +5,79 @@
 use Magento\Framework\Event\ObserverInterface;
 use Magento\Framework\MessageQueue\PublisherInterface;
 use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Sales\Model\ResourceModel\Order as OrderResource;
+use StoreKeeper\StoreKeeper\Api\PaymentApiClient;
 use StoreKeeper\StoreKeeper\Helper\Api\Auth;
 
 class SalesOrderInvoicePayObserver implements ObserverInterface
 {
     private Auth $authHelper;
     private Json $json;
     private PublisherInterface $publisher;
+    private PaymentApiClient $paymentApiClient;
+    private OrderResource $orderResource;
 
     /**
      * Constructor
      *
      * @param Auth $authHelper
      * @param Json $json
      * @param PublisherInterface $publisher
+     * @param PaymentApiClient $paymentApiClient
+     * @param OrderResource $orderResource
      */
     public function __construct(
         Auth $authHelper,
         Json $json,
-        PublisherInterface $publisher
+        PublisherInterface $publisher,
+        PaymentApiClient $paymentApiClient,
+        OrderResource $orderResource
     ) {
         $this->authHelper = $authHelper;
         $this->json = $json;
         $this->publisher = $publisher;
+        $this->paymentApiClient = $paymentApiClient;
+        $this->orderResource = $orderResource;
     }
 
     /**
-     * Set order as pending for sync
+     * Sync invoice creation for non-SK payment methods
      *
      * @param Observer $observer
      * @return void
      */
     public function execute(Observer $observer)
     {
         $invoice = $observer->getEvent()->getInvoice();
-        $order = $invoice->getOrder();
+        if ($invoice->getState() == \Magento\Sales\Model\Order\Invoice::STATE_PAID) {
+            $order = $invoice->getOrder();
+            $storekeeperId = $order->getStorekeeperId();
+            if (
+                $this->authHelper->isConnected($order->getStoreId())
+                && !$this->paymentApiClient->isStorekeeperPayment($order->getPayment()->getMethod())
+                && $this->authHelper->isOrderSyncEnabled($order->getStoreId())
+                && !$order->getOrderDetached()
+                && $storekeeperId
```

Comment:
what if it was paid before it was synchronized to storekeeper? was this case handled somewhere else?